### PR TITLE
[language][move] Added naming tests

### PR DESCRIPTION
**Pull request by tnowacki**
## Motivation

- Added tests for naming pass.
- Tried to cover +/- cases for each check

## Test Plan

ran em

### CODE REVIEW

File: language/move-lang/tests/move_check/naming/duplicate_acquires_list_item.move [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module M {
+    resource struct R {}
+    resource struct X {}
+    foo() acquires R, X, R {
```
Comment (tzakian) [inline language/move-lang/tests/move_check/naming/duplicate_acquires_list_item.move]: What do you think of also having a test for multiple duplicate acquires items (e.g. `R, X, R, R`). And making sure this produces the errors expected.

File: language/move-lang/tests/move_check/naming/duplicate_type_parameter_function.exp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+   │
+
+error: 
+
+   ┌── tests/move_check/naming/duplicate_type_parameter_function.move:3:23 ───
+   │
+ 3 │     foo2<T: copyable, T: resource, T>() {}
+   │                                    ^ Duplicate type parameter declared with name 'T'
+   ·
+ 3 │     foo2<T: copyable, T: resource, T>() {}
```
Comment (tzakian) [inline language/move-lang/tests/move_check/naming/duplicate_type_parameter_function.exp]: Should the error be pointing at this `T`? Why not the first definition of the `T`?

Comment (tnowacki) [reply language/move-lang/tests/move_check/naming/duplicate_type_parameter_function.exp]: I could.... Currently the unique map updates the loc on the next definition. So the second duplicate points to the first def, and the third points to the second. 

My thoughts was this keeps errors a bit more contextual and close together, but maybe its too confusing? Thoughts?
